src/arrayMethodSort.js

```diff
@@ -5,7 +5,47 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (this.length < 2) {
+      return this;
+    }
+
+    const isFunc = compareFunction !== undefined ? 1 : 0;
+    let res = [];
+
+    const startItem = this[0];
+    const lessThenStart = [];
+    const biggerThenStart = [];
+    const equalStart = [startItem];
+
+    for (let i = 1; i < this.length; i++) {
+      if (!isFunc) {
+        if (startItem.toString() > this[i].toString()) {
+          lessThenStart.push(this[i]);
+        } else if (startItem.toString() < this[i].toString()) {
+          biggerThenStart.push(this[i]);
+        } else {
+          equalStart.push(this[i]);
+        }
+      } else {
+        if (compareFunction(startItem, this[i]) > 0) {
+          lessThenStart.push(this[i]);
+        } else if (compareFunction(startItem, this[i]) < 0) {
+          biggerThenStart.push(this[i]);
+        } else {
+          equalStart.push(this[i]);
+        }
+      }
+    }
```

Comment:
to be honest, it doesn't have any sense to create two arrays and duplicate logic inside if/else block.
you can make a simple bubble sorting instead. check [this](https://www.geeksforgeeks.org/bubble-sort-algorithms-by-using-javascript/)

Reply:
Thanks a lot. I rewrote the function using bubble sort